### DISCUSSION TITLE
Attachments with URL in Email API

### DISCUSSION THREAD

**Discussion by rccess** [postalserver/postal#2375] [Help with using Postal]
Hello team,

I am trying to send email through email api with attachment in url ,
But is not picking the attachment , email sent but attachment not attached 

is it right way to do or any other way to attach docs through url in email api 

------------------------------
{
	"to": ["abc@gmail.com"],
	"from": "Company<info@xyz.com>",
	"sender": "Company",
	"subject": "Email API Test",
	"plain_body": "hello this email api checking",
	"attachments": ["http://xyz.com/images/article/444242442.docx"]
}

**Comment by willpower232** [comment]
It is not very clear in the current documentation https://apiv1.postalserver.io/controllers/send/message.html

but referencing a client library, you can see attachments should be an array of arrays which includes the base64 encoded file contents

https://github.com/postalserver/postal-php/blob/7656516590fe1a409b94833000e14fcf87c61c72/src/SendMessage.php#L75-L84

**Comment by rccess** [comment]
i have checked that already 

actually i am try in curl but i don't know how to define path in curl request 

can you please help me with the same

**Comment by rccess** [comment]
Fixed and working

**Comment by rccess** [comment]
Hello @willpower232 

attachment picked successfully but file got courrepted 

i have tested with word, pdf , txt file. all are receiving as a corrupted files,

here is the code i am using
for txt
--------------------------------------
{
    "to": ["abc@gmail.com"
    ],
    "from":"company<info@xyz.com>", 
    "sender": "company", 
    "subject": "Email API Test",
    "plain_body": "hello this email api checking",
    "attachments": [{ 
      "name": "docs.txt", 
      "content_type": "text/plain", 
      "data": "base64_encode($data)", 
      "content_url": "https://www.xyz.com/wp-content/uploads/2023/03/docs.txt"
    }]
}

-------------------------------------
for docx
{
    "to": ["abc@gmail.com"
    ],
    "from":"company<info@xyz.com>", 
    "sender": "company", 
    "subject": "Email API Test",
    "plain_body": "hello this email api checking",
    "attachments": [{ 
      "name": "123.docx", 
      "content_type": "application/msword", 
      "data": "base64_encode($data)", 
      "content_url": "http://domain.com/images/article/123.docx"
    }]
}

what might be the problem ?

**Reply by willpower232** [reply]
`content_url` would not be used so you don't need to bother with that

you should double check that `$data` is the real file contents, for plain text you could even try `$data = 'this is some plain text'`

**Comment by rccess** [comment]
i have tried this also but issue is the same 

{
    "to": ["abc@gmail.com"
    ],
    "from":"company <info@xyz.com>", 
    "sender": "company", 
    "subject": "Email API Test",
    "plain_body": "hello this email api checking",
    "attachments": [{ 
      "name": "123.pdf", 
      "content_type": "application/pdf", 
      "data": "base64_encode(file_get_content('https://domain.com/123.pdf'))"
    }]
}

can you please advise what is missing there ?

**Reply by willpower232** [reply]
I would imagine that `file_get_contents` failed so you didn't get any content. You should verify you have received data before trying to send a message

**Reply by rccess** [reply]
actually i am try to send from postman so i don't know how to print to check whats is returning in data

can you please help me in the same,
can you please help me with json code what should put in code to check return data

<img width="512" alt="image" src="https://user-images.githubusercontent.com/32220165/225677183-117502bd-5fe0-4764-91ca-f43adb5d8bed.png">

**Reply by willpower232** [reply]
Unfortunately postman cannot execute PHP code so that explains why it is not working. To complete your test, you basically need to do something like this

```
"attachments": [{
"name": "docs.txt",
"content_type": "text/plain",
"data": "hello I am a plain text file",
}]
```
